### PR TITLE
CA-401242: avoid long-running, idle connections on VDI.pool_migrate

### PR DESCRIPTION
**Pull request by psafont**
When a VDI.pool_migrate starts at a pool member, a connection between the
coordinator and that host remains open for the duration of the migration. This
connection is completely idle. If it's open for 12 hours, stunnel closes the
connection due to inactivity, which cancels the migration.

To avoid this use an internal API that uses short-running connection whenever
possible to avoid interrupting the migration.

Also change nested if-else flow in vdi handling to use Result with let binds, and add missing CDI operations to storage operations.

I've manually verified the fix by changing the stunnel timeout in the coordinator of a pool to 5 seconds and migrating a vdi used by a VM in another host:
```
# xe vdi-pool-migrate uuid=fdb8783f-8cdb-44d3-9327-992f047e6262 sr-uuid=c879ffc9-bb14-fec5-3d95-8959db9337eb
270c3dc8-27bc-4ef2-982a-a6142f0ffce1
```

And after reverting the patch:
```
# xe vdi-pool-migrate uuid=270c3dc8-27bc-4ef2-982a-a6142f0ffce1 sr-uuid=aade9dc6-ff31-3204-4187-37f82ad06c77
Cannot forward messages because the server cannot be contacted. The server may be switched off or there may be network connectivity problems.
host: 0a3a45a0-8754-4570-bfde-6ef6843ccda1 (xs2)
```

The changes are best reviewed one by one, and while ignoring whitespace, if using github.

### CODE REVIEW

**Comment by last-genius** [comment]
might be worth specifying that the connection that's idle for 12 hours is not the main one, but the auxiliary one

**Review by lindig — APPROVED** [review]

**Review by robhoes — APPROVED** [review]

File: ocaml/xapi/xapi_vdi.ml [hunk 1]
```diff
@@ -22,49 +22,49 @@ open D
 (**************************************************************************************)
 (* current/allowed operations checking                                                *)
 
+let feature_of_op =
```
Comment (Vincent-lau) [inline ocaml/xapi/xapi_vdi.ml:25]: IMO this function should go to `Smint` but I think it is ok to keep it here for this PR for clarity as it was here.

File: ocaml/tests/record_util/old_record_util.ml [hunk 2]
```diff
@@ -341,6 +341,21 @@ let sr_operation_to_string : API.storage_operations -> string = function
       "PBD.create"
   | `pbd_destroy ->
       "PBD.destroy"
+  (* The following ones were added after the file got introduced *)
+  | `vdi_blocked ->
+      "VDI.copy"
```
Comment (Vincent-lau) [inline ocaml/tests/record_util/old_record_util.ml:346]: should this be `VDI.blocked`?

Comment (psafont) [reply ocaml/tests/record_util/old_record_util.ml]: yup!

**Review by Vincent-lau — APPROVED** [review]

**Pull request merged**